File: apps/api/src/moldo_api/schemas/exam.py
from typing import Literal

from pydantic import BaseModel, ConfigDict


ExamMode = Literal["mock"]
ExamIconName = Literal["document", "message", "target"]
QuestionType = Literal[
    "self_intro",
    "personal_prompt",
    "role_play",
    "past_experience",
    "comparison",
    "problem_solving",
]


class ApiDto(BaseModel):
    model_config = ConfigDict(extra="forbid")


class ExamSummary(ApiDto):
    id: str
    title: str
    tag: str
    questionCount: int
    estimatedMinutes: int
    mode: ExamMode
    icon: ExamIconName
    description: str | None = None


class ExamQuestion(ApiDto):
    id: str
    examId: str
    order: int
    type: QuestionType
    ttsScriptEn: str
    ttsAudioUrl: str | None = None
    prepSeconds: int
    answerSeconds: int

File: packages/types/src/exam.ts
export type ExamMode = "mock";

export type ExamIconName = "document" | "message" | "target";

export type ExamSummary = {
  id: string;
  title: string;
  tag: string;
  questionCount: number;
  estimatedMinutes: number;
  mode: ExamMode;
  icon: ExamIconName;
  description?: string;
};

export type QuestionType =
  | "self_intro"
  | "personal_prompt"
  | "role_play"
  | "past_experience"
  | "comparison"
  | "problem_solving";

export type ExamQuestion = {
  id: string;
  examId: string;
  order: number;
  type: QuestionType;
  ttsScriptEn: string;
  ttsAudioUrl?: string;
  prepSeconds: number;
  answerSeconds: number;
};
